src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -685,25 +685,17 @@ private void LoadAssemblies(IEnumerable<string> assemblies)
         /// </summary>
         private static IEnumerable<PortableExecutableReference> InitDefaultRefAssemblies()
         {
-            // Define number of reference assemblies distributed with PowerShell.
-            const int maxPowershellRefAssemblies = 161;
+            var dotnetRefAsmFiles = Directory.EnumerateFiles(s_netcoreAppRefFolder, "*.dll", SearchOption.TopDirectoryOnly);
```

Comment:
This will enumerate the directory twice, once for the `dotnetRefAsmFiles.Count()` and then again in the `foreach` loop.  Can we just create the list initially at some fixed size and then let it re-allocate as needed?  Is that such a big cost?  If so, then maybe we should just have a temporary List of files, which seems better than iterating over the directory twice.

Reply:
We wouldn't start at zero, but at 161.  Only one additional allocation would likely be needed.

Reply:
It's an interesting problem to think about (I took a look at implementation of some .NET APIs and that was good learning), but not good if it took too much of our time. I have reverted to use 200 as the initial capacity and let it auto-expand if needed (should be unlikely).

Reply:
Sounds good.  Sorry for being a stickler, but I feel this better.

Reply:
200 certainly provides headroom for the foreseeable future